x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptDoubleMappedFieldType.java

```diff
@@ -0,0 +1,103 @@
+import org.elasticsearch.xpack.runtimefields.query.DoubleScriptFieldTermQuery;
+import org.elasticsearch.xpack.runtimefields.query.DoubleScriptFieldTermsQuery;
+
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Map;
+
+public class ScriptDoubleMappedFieldType extends AbstractScriptMappedFieldType {
+    private final DoubleScriptFieldScript.Factory scriptFactory;
+
+    ScriptDoubleMappedFieldType(String name, Script script, DoubleScriptFieldScript.Factory scriptFactory, Map<String, String> meta) {
+        super(name, script, meta);
+        this.scriptFactory = scriptFactory;
+    }
+
+    @Override
+    protected String runtimeType() {
+        return NumberType.DOUBLE.typeName();
+    }
+
+    @Override
+    public Object valueForDisplay(Object value) {
+        return value; // These should come back as a Double
+    }
+
+    @Override
+    public ScriptDoubleFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName) {
+        // TODO once we get SearchLookup as an argument, we can already call scriptFactory.newFactory here and pass through the result
+        return new ScriptDoubleFieldData.Builder(script, scriptFactory);
+    }
+
+    private DoubleScriptFieldScript.LeafFactory leafFactory(QueryShardContext context) {
+        return scriptFactory.newFactory(script.getParams(), context.lookup());
+    }
+
+    @Override
+    public Query existsQuery(QueryShardContext context) {
+        checkAllowExpensiveQueries(context);
+        return new DoubleScriptFieldExistsQuery(script, leafFactory(context), name());
+    }
+
+    @Override
+    public Query rangeQuery(
+        Object lowerTerm,
+        Object upperTerm,
+        boolean includeLower,
+        boolean includeUpper,
+        ShapeRelation relation,
+        ZoneId timeZone,
+        DateMathParser parser,
+        QueryShardContext context
+    ) {
+        checkAllowExpensiveQueries(context);
+        return NumberType.doubleRangeQuery(
+            lowerTerm,
+            upperTerm,
+            includeLower,
+            includeUpper,
+            (l, u) -> new DoubleScriptFieldRangeQuery(script, leafFactory(context), name(), l, u)
+        );
+    }
+
+    @Override
+    public Query termQuery(Object value, QueryShardContext context) {
+        checkAllowExpensiveQueries(context);
+        return new DoubleScriptFieldTermQuery(script, leafFactory(context), name(), NumberType.objectToDouble(value));
+    }
+
+    @Override
+    public Query termsQuery(List<?> values, QueryShardContext context) {
+        if (values.isEmpty()) {
+            return Queries.newMatchAllQuery();
+        }
+        LongSet terms = new LongHashSet(values.size());
+        for (Object value : values) {
+            terms.add(Double.doubleToLongBits(NumberType.objectToDouble(value)));
+        }
+        checkAllowExpensiveQueries(context);
+        return new DoubleScriptFieldTermsQuery(script, leafFactory(context), name(), terms);
+    }
```

Comment:
I was wondering, would it make sense to add tests for the unsupported queries, to verify that they throw error?

Reply:
I can do that!